TreasureData.agent.lib.nut

```diff
@@ -0,0 +1,144 @@
+    _debug = null;
+
+    // TreasureData class constructor.
+    //
+    // Parameters:
+    //     apiKey : string           Treasure Data API key.
+    //                               See https://support.treasuredata.com/hc/en-us/articles/360000763288-Get-API-Keys
+    //
+    // Returns:                      TreasureData instance created.
+    constructor(apiKey) {
+        _apiKey = apiKey;
+        _sendHeaders = {
+            "Content-Type" : "application/json",
+            "X-TD-Write-Key" : _apiKey
+        };
+    }
+
+    // Sends data record to the specified Treasure Data table.
+    // See https://support.treasuredata.com/hc/en-us/articles/360000675487-Postback-API#POST%20%2Fpostback%2Fv3%2Fevent%2F%7Bdatabase%7D%2F%7Btable%7D
+    //
+    // Parameters:
+    //     dbName : string           Treasure Data database name.
+    //     tableName : string        Treasure Data table name.
+    //     data : table              The data to send - a key-value table,
+    //                               where a "key" is the name of the field (column) in the database table,
+    //                               the "value" is the value to be written into that field.
+    //     callback : function       Optional callback function executed once the operation
+    //         (optional)            is completed.
+    //                               The callback signature:
+    //                               callback(error, data), where
+    //                                   error : table               Error details, HTTP response table (see https://electricimp.com/docs/api/httprequest/sendasync/)
+    //                                                               or null if the operation succeeds.
+    //                                   data : table                The original data passed to 
+    //                                                               sendData() method.
+    function sendData(dbName, tableName, data, callback = null) {
+        _processRequest(
+            "POST",
+            _TREASURE_DATA_SEND_ENDPOINT,
+            format(_TREASURE_DATA_SEND_PATH, dbName, tableName),
+            _sendHeaders,
+            data,
+            function (error) {
+                _invokeDefaultCallback(callback, error, data);
+            }.bindenv(this));
+    }
+
+    // Enables/disables the library debug output (including errors logging).
+    // Disabled by default.
+    //
+    // Parameters:
+    //     value : boolean             true to enable, false to disable
+    function setDebug(value) {
+        _debug = value;
+    }
+
+    // -------------------- PRIVATE METHODS -------------------- //
+
+    // Sends an http request to TreasureData.
+    function _processRequest(method, endpoint, path, headers, body, callback) {
+        local url = endpoint + path;
+        local encodedBody = http.jsonencode(body);
+        _logDebug(format("Doing the request: %s %s, body: %s", method, url, encodedBody));
```

Comment:
I would phrase this log msg to something like this "Sending %s request to %s with body: %s"